packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2124,34 +2087,40 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
       return border;
     }
 
-    final Color borderColor;
-    if (decoration.enabled || isFocused) {
-      borderColor = decoration.errorText == null
-        ? _getDefaultBorderColor(themeData)
-        : themeData.errorColor;
-    } else {
-      borderColor = ((decoration.filled ?? false) && !(decoration.border?.isOutline ?? false))
-        ? Colors.transparent
-        : themeData.disabledColor;
+    if (themeData.useMaterial3) {
+      if (decoration.filled!) {
+        return border.copyWith(
+          borderSide: MaterialStateProperty.resolveAs(defaults.activeIndicatorBorder, materialState),
+        );
+      }
+      else {
```

Comment:
Nit: the else should be on the same line with the }

Reply:
Done